docs/programming_guide.rst

```diff
@@ -0,0 +1,83 @@
+supported by HIP is outlined in this chapter. In general, GPUs are made up of
+many so called Compute Units that excel at executing parallelizable,
+computationally intensive workloads without complex control-flow.
+
+Increase parallelism on multiple level
+================================================================================
+
+To maximize performance and keep all system components fully utilized, the
+application should expose and efficiently manage as much parallelism as possible.
+:ref:`Parallel execution <parallel execution>` can be achieved at the
+application, device, and multiprocessor levels.
+
+The application’s host and device operations can achieve parallel execution
+through asynchronous calls, streams, or HIP graphs. On the device level,
+multiple kernels can execute concurrently when resources are available, and at
+the multiprocessor level, developers can overlap data transfers with
+computations to further optimize performance.
+
+Memory management
+================================================================================
+
+GPUs generally have their own destinct memory, also referred to as :ref:`device
+memory <device_memory>`, separate from the :ref:`host memory <device_memory>`.
+This memory needs to be managed separately from the host memory. This includes
+allocating the memory and transfering it between the host and the device. These
+operations can be performance critical, so it is important to know how to use
+them effectively.
+
+For more information, see :ref:`memory management <memory_management>`.
+
+Synchronize CPU and GPU Workloads
+================================================================================
+
+Tasks on the host and devices run asynchronously, so proper synchronization is
+needed when dependencies between those tasks exist. Asynchronous execution of
+tasks is useful to fully utilize the available resources. Even when only a
+single device is available, memory transfers and the execution of tasks can be
+overlapped with asynchrononus execution.
+
+Error Handling
+================================================================================
+
+All functions in the HIP runtime API return an error value of type
+:cpp:enum:`hipError_t`, that can be used to verify, whether the function was
+successfully executed. It is therefore of utmost importance, to check these
+returned values, in order to catch and handle those errors, if possible.
+An exception to this are kernel launches, which do not return any value. These
+errors can be caught with specific functions like :cpp:func:`hipGetLastError()`.
+
+Multi-GPU and Load Balancing
+================================================================================
+
+Large-scale applications that need more compute power can use multiple GPUs in
+the system. This requires distributing workloads across multiple GPUs to balance
+the load to prevent some GPUs from being overutilized while others are idle.
```

Comment:
```suggestion
the load to prevent some GPUs from being overutilized while others are idle. For more information, see XXX. 
```

Reply:
This topic is not written yet. We will link in the PR of the new section.